drivers/timer/Kconfig.mcux_lptmr

```diff
@@ -6,11 +6,22 @@
 config MCUX_LPTMR_TIMER
 	bool "MCUX LPTMR timer"
 	default y
-	depends on DT_HAS_NXP_KINETIS_LPTMR_ENABLED
+	depends on DT_HAS_NXP_KINETIS_LPTMR_ENABLED || \
+		   MCUX_KINETIS_LPTMR
 	depends on !COUNTER_MCUX_LPTMR
 	depends on PM
 	select SYSTEM_TIMER_HAS_DISABLE_SUPPORT
 	help
 	  This module implements a kernel device driver for the NXP MCUX Low
 	  Power Timer (LPTMR) and provides the standard "system clock driver"
 	  interfaces.
+
+config MCUX_KINETIS_LPTMR
+	bool "Deprecated DT compatible"
```

Comment:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/34665051/064d4d1d-5181-4d16-9164-bf9bd2192a5a)
This is how this appears in menuconfig. I think we should choose a more descriptive name for this boolean, perhaps "MCUX LPTMR timer [DEPRECATED]"?

For reference, this is another deprecated Kconfig in the tree: 
![image](https://github.com/zephyrproject-rtos/zephyr/assets/34665051/e72821bd-2521-48ee-bda0-9b5e30009356)

Reply:
I will fix this, also for the record my opinion is we should just break the downstream boards, since they can fix it very quickly, and not do silly transitions for tiny things like this